misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -32,10 +33,13 @@ internal class RealTransacter private constructor(
 
   constructor(
     qualifier: KClass<out Annotation>,
-    sessionFactory: SessionFactory,
+    sessionFactoryProvider: Provider<SessionFactory>,
     config: DataSourceConfig,
     tracer: Tracer?
-  ) : this(qualifier, sessionFactory, config, ThreadLocal(), TransacterOptions(), tracer)
+  ) : this(qualifier, sessionFactoryProvider, config, ThreadLocal(), TransacterOptions(), tracer)
+
+  private val sessionFactory
+    get() = sessionFactoryProvider.get()
```

Comment:
This is only run when the `sessionFactory` property is accessed. It's like a getter.